recipes/date/all/conanfile.py

```diff
@@ -44,29 +43,30 @@ def config_options(self):
             self.options.use_system_tz_db = True
 
     def configure(self):
-        if self.options.shared:
-            del self.options.fPIC
+        self.package_type = "header-library" if self.options.header_only else "library"
```

Comment:
Please do not do this :) it's not intended 

https://github.com/conan-io/conan/issues/13145

```suggestion
```

Reply:
You refer to a different problem and a different kind of recipe. It's still unclear if it's the right thing to do. See this discussion: https://github.com/conan-io/conan/issues/13358

Reply:
Unclear == don't do it :)

So CCI doesn't conform to the indented solution.

The design is a project is compiled or header not both. So those two methods should be separate recipes.

But since the client understands all three options its more then smart enough to determine the package type.

So there there should be _no_ package type and it will do the right thing 

Package type is optional should be omitted  as much as possible (the main use case is recipes with no options)

Reply:
Is it conan team consensus or just yours? When I read https://github.com/conan-io/conan/issues/13358, memshared doesn't exactly say what you are saying.